llvm/include/llvm/CodeGen/GlobalISel/LegalizerInfo.h

```diff
@@ -926,6 +926,16 @@ class LegalizeRuleSet {
                                      Types2);
   }
 
+  /// The instruction is custom when the predicate is true and type indexes 0
+  /// and 1 are all in their respective lists.
+  LegalizeRuleSet &
+  customForCartesianProduct(bool Pred, std::initializer_list<LLT> Types0,
```

Comment:
@arsenm what do you think, should we add `Pred` version for all the overloads of `customForCartesianProduct`?

Reply:
Eventually probably should have both, can add as needed